scylla-cql/src/types/serialize/row.rs

```diff
@@ -200,3 +206,127 @@ mod tests {
         assert_eq!(sorted_row_data, unsorted_row_data);
     }
 }
+
+/// Keeps a buffer with serialized Values
+/// Allows adding new Values and iterating over serialized ones
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct NewSerializedValues {
+    serialized_values: Vec<u8>,
+}
+
+impl NewSerializedValues {
+    pub fn new() -> Self {
+        let mut buf = Vec::new();
+        buf.extend_from_slice(&0u16.to_be_bytes());
```

Comment:
It's a little sad that `NewSerializedValues::new()` must allocate - this prevents us from making the method `const` and forces us to allocate in cases where we need to pass an empty list of values.

Perhaps we could rework how `BufBackedRowWriter` works and let it count written values in a variable and not in the buffer.

No need to act upon it now, this can be left for a later improvement.